compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1021,9 +1021,83 @@ TR::Register *OMR::Power::TreeEvaluator::mxorEvaluator(TR::Node *node, TR::CodeG
     return TR::TreeEvaluator::vxorEvaluator(node, cg);
 }
 
+static TR::Register *mloadiFromArrayHelper(TR::Node *node, TR::CodeGenerator *cg,
+    TR::InstOpCode::Mnemonic reverseLoadOp, TR::InstOpCode::Mnemonic loadOp, TR::InstOpCode::Mnemonic subOp,
+    int dataSize)
+{
+    TR::Node *child = node->getFirstChild();
+
+    TR::Register *dstReg = cg->allocateRegister(TR_VRF);
+    TR::Register *zeroReg = cg->allocateRegister(TR_VRF);
+    TR::Register *tmpReg = NULL;
+
+    node->setRegister(dstReg);
+
+    auto ref = TR::LoadStoreHandlerImpl::generateMemoryReference(cg, node, dataSize, true, 0);
+
+    // Due to the semantics of the vector unpack instructions, we can only use vector load for Byte/ShortVector masks
+    if (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P9) && (dataSize >= 8))
```

Comment:
it looks wrong to me with `dataSize >= 8`. it should be strictly `dataSize > 8`

Reply:
`vupkhsb` only looks at the top half of the contents of the input register (`dstReg` in this case), so in the case where `dataSize == 8` (i.e.: ShortVector mask), I believe we should be able to use `lxvb16x` safely to get the 8 bytes worth of data from the input boolean array without having to worry about endianness (since it is endian agnostic) or having to take an extra instruction to move the data into a vector register.  The additional 8 bytes that are loaded into the bottom half of `dstReg` will go unused:
<img width="835" height="147" alt="image" src="https://github.com/user-attachments/assets/f6cf87f4-7cb3-47dc-bff1-2f7498935c96" />
(taken from the Power ISA)